### PR TITLE
[Merged by Bors] - chore: add some Unique instances

### PR DESCRIPTION
**Pull request by alreadydone**
The aim is to remove some extraneous Nonempty assumptions in Algebra/DirectLimit.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alreadydone** [comment]
!bench

**Comment by alreadydone** [comment]
now in #8473

**Comment by alreadydone** [comment]
Because Logic/Unique's dependency is lighter after #8510, I now import it into Data/Quot in favor of [easier constructions](https://github.com/leanprover-community/mathlib4/pull/8500/files#diff-0d878a8f283a3ea761f136a02767f65ed7459edbc00f79f7ab9e86e93e954a0dR8-R236) of the Unique instances using the Subsingleton and Inhabited instances there.

File: Mathlib/Algebra/DirectLimit.lean [hunk 1]
```diff
@@ -103,6 +103,9 @@ instance module : Module R (DirectLimit G f) :=
 instance inhabited : Inhabited (DirectLimit G f) :=
   ⟨0⟩
 
+instance unique [IsEmpty ι] : Unique (DirectLimit G f) := by
+  change Unique (Quotient _); infer_instance
```
Comment (eric-wieser) [inline Mathlib/Algebra/DirectLimit.lean:107]: or alternatively:
```suggestion
instance unique [IsEmpty ι] : Unique (DirectLimit G f) :=
  inferInstanceAs <| Unique (Quotient _)
```

File: Mathlib/SetTheory/Cardinal/Ordinal.lean [hunk 2]
```diff
@@ -1071,7 +1071,6 @@ theorem mk_multiset_of_nonempty (α : Type u) [Nonempty α] : #(Multiset α) = m
 theorem mk_multiset_of_infinite (α : Type u) [Infinite α] : #(Multiset α) = #α := by simp
 #align cardinal.mk_multiset_of_infinite Cardinal.mk_multiset_of_infinite
 
-@[simp]
```
Comment (eric-wieser) [inline Mathlib/SetTheory/Cardinal/Ordinal.lean:1074]: I guess the linter now complains because this follows by uniqueness?

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

File: Mathlib/SetTheory/Cardinal/Ordinal.lean (hunk 2 above)
Comment (alreadydone) [reply Mathlib/SetTheory/Cardinal/Ordinal.lean:1074]: Yeah, as shown by [this CI run](https://github.com/leanprover-community/mathlib4/actions/runs/6916053475/job/18815681354).

**Pull request closed without merging**